File: ruby/lasagna/lasagna.rb
class Lasagna
  EXPECTED_MINUTES_IN_OVEN = 40

  def remaining_minutes_in_oven(actual_minutes_in_oven)
    return EXPECTED_MINUTES_IN_OVEN - actual_minutes_in_oven
  end

  def preparation_time_in_minutes(layers)
    return 2 * layers
  end

  def total_time_in_minutes(number_of_layers:, actual_minutes_in_oven:)
    puts "#{number_of_layers} layers, #{actual_minutes_in_oven} minutes in oven"

    prep_time = preparation_time_in_minutes(number_of_layers)
    remaining_time = remaining_minutes_in_oven(actual_minutes_in_oven)

    puts "Preparation time: #{prep_time} minutes, Atual time: #{actual_minutes_in_oven} minutes"
    puts "Total time: #{prep_time + remaining_time} minutes"
    
    return prep_time + actual_minutes_in_oven
  end
end

lasagna = Lasagna.new
lasagna.total_time_in_minutes(number_of_layers: 3, actual_minutes_in_oven: 20)

File: java/lasagna/src/main/java/Lasagna.java
public class Lasagna {
    private int expectedMinutesInOven = 40,
                prepTimePerLayer = 2;
    
    /**
     * Get expected amount of minutes in the oven.
     * @return expected minutes.
     */
    public int expectedMinutesInOven(){
        return expectedMinutesInOven;
    }
    
    /**
     * Calculate remaining minutes in the oven.
     * @param elapsedMinutes time already in the oven.
     * @return remaining time in the oven.
     */
    public int remainingMinutesInOven(int elapsedMinutes){
        return expectedMinutesInOven - elapsedMinutes;
    }

    /**
     * Calculate preparation time for a given number of
     * layers.
     * @param layers number of layers.
     * @return preparation time in minutes.
     */
    public int preparationTimeInMinutes(int layers){
        return prepTimePerLayer * layers;
    }

    /**
     * Calculate total time given a number of layers and
     * elapsed minutes in the oven.
     * @param layers number of layers.
     * @param elapsedMinutes minutes in the oven.
     * @return Total time taken.
     */
    public int totalTimeInMinutes(int layers, int elapsedMinutes){
        return preparationTimeInMinutes(layers) + elapsedMinutes;
    }
}